CMakeLists.txt

```diff
@@ -13,8 +13,16 @@ endif ()
 include(cmake/compilation_flags.cmake)
 list(APPEND GLOBAL_CXXFLAGS ${DEFAULT_CXX_FLAGS})
 
+option(BUILD_SHARED_LIBS "Build Shared Libraries" OFF)
```

Comment:
This setting will not work when compiling with Visual Studio, because the corresponding import library (pe-parser-library.lib) is not generated when building the shared library.

We should either disable this setting when compiling with MSVC or add support for generating and installing the required .lib file.

Reply:
I don't have an MSVC system to test, but cmake automatically creates the *.dll.a file - I thought this was the mingw equivalent of *.lib files? I'll do some reading.

Reply:
It seems difference is that MSVC doesn't export all symbols by default, so the import lib isn't created. There's a newish ["export all" feature in CMake 3.4](https://blog.kitware.com/create-dlls-on-windows-without-declspec-using-new-cmake-export-all-feature/) that I've conditionally enabled.

Tested on an MSVC system (with some new .gitignore entries) and [all seems to be working as expected](https://gist.github.com/tonytheodore/18fce7a26fbb7454bd5e90db1526bd34#file-x86-release-shared-L123-L124).

Reply:
Thanks for looking into this issue!

Reply:
You're welcome - thanks for such a useful library! I found it indirectly via https://github.com/gsauthof/pe-util.